cairo/ethereum/cancun/fork.cairo

```diff
@@ -1009,52 +1033,80 @@ func apply_body{
-    // beacon_block_roots_contract_code = get_account(
-    //     state, BEACON_ROOTS_ADDRESS
-    // ).code
-
-    // system_tx_message = Message(
-    //     caller=SYSTEM_ADDRESS,
-    //     target=BEACON_ROOTS_ADDRESS,
-    //     gas=SYSTEM_TRANSACTION_GAS,
-    //     value=U256(0),
-    //     data=parent_beacon_block_root,
-    //     code=beacon_block_roots_contract_code,
-    //     depth=Uint(0),
-    //     current_target=BEACON_ROOTS_ADDRESS,
-    //     code_address=BEACON_ROOTS_ADDRESS,
-    //     should_transfer_value=False,
-    //     is_static=False,
-    //     accessed_addresses=set(),
-    //     accessed_storage_keys=set(),
-    //     parent_evm=None,
-    // )
-
-    // system_tx_env = vm.Environment(
-    //     caller=SYSTEM_ADDRESS,
-    //     origin=SYSTEM_ADDRESS,
-    //     block_hashes=block_hashes,
-    //     coinbase=coinbase,
-    //     number=block_number,
-    //     gas_limit=block_gas_limit,
-    //     base_fee_per_gas=base_fee_per_gas,
-    //     gas_price=base_fee_per_gas,
-    //     time=block_time,
-    //     prev_randao=prev_randao,
-    //     state=state,
-    //     chain_id=chain_id,
-    //     traces=[],
-    //     excess_blob_gas=excess_blob_gas,
-    //     blob_versioned_hashes=(),
-    //     transient_storage=TransientStorage(),
-    // )
-
-    // system_tx_output = process_message_call(system_tx_message, system_tx_env)
-
-    // destroy_touched_empty_accounts(
-    //     system_tx_env.state, system_tx_output.touched_accounts
-    // )
+    tempvar beacon_roots_address = Address(BEACON_ROOTS_ADDRESS);
+    let beacon_roots_account = get_account{state=state}(beacon_roots_address);
+    let beacon_block_roots_contract_code = beacon_roots_account.value.code;
+
+    let data = Bytes32_to_Bytes(parent_beacon_block_root);
+    let code_address = OptionalAddress(&beacon_roots_address);
+
+    // empty access addresses with default_dict_new(0)
+    let (empty_data_ptr) = default_dict_new(0);
+    tempvar accessed_addresses = SetAddress(
+        new SetAddressStruct(
+            dict_ptr_start=cast(empty_data_ptr, SetAddressDictAccess*),
+            dict_ptr=cast(empty_data_ptr, SetAddressDictAccess*),
+        ),
+    );
+
+    // empty access storage keys with default_dict_new(0)
+    let (empty_data_ptr) = default_dict_new(0);
+    tempvar accessed_storage_keys = SetTupleAddressBytes32(
+        new SetTupleAddressBytes32Struct(
+            dict_ptr_start=cast(empty_data_ptr, SetTupleAddressBytes32DictAccess*),
+            dict_ptr=cast(empty_data_ptr, SetTupleAddressBytes32DictAccess*),
+        ),
+    );
+    tempvar system_tx_message = Message(
+        new MessageStruct(
+            caller=Address(SYSTEM_ADDRESS),
+            target=To(new ToStruct(bytes0=cast(0, Bytes0*), address=&beacon_roots_address)),
+            current_target=beacon_roots_address,
+            gas=Uint(SYSTEM_TRANSACTION_GAS),
+            value=U256(new U256Struct(0, 0)),
```

Comment:
use `U256_ZERO` from `state.cairo` (which might need to be moved to `constants.cairo`)

Reply:
you can just do cast(U256_ZERO, U256Struct*)

Reply:
ok thanks will do, waiting on CI first to pass
get_label_location is not used in fork.cairo though